plugin/core/file_watcher.py

```diff
@@ -0,0 +1,76 @@
+from .typing import List, Optional, Protocol, Tuple, Type
+from abc import ABCMeta
+from abc import abstractmethod
+
+
+DEFAULT_IGNORES = ['**/.git/**', '**/node_modules/**', '**/.hg/**']
+
+WatchKind = int
+FilePath = str
+FileEvent = Tuple[WatchKind, FilePath]
+
+
+class WatchKindValue:
+    CREATE = 1
+    CHANGE = 2
+    DELETE = 4
```

Comment:
Why didn't you use `3` (by the spec) instead of `4` for the DELETE?

```
export namespace FileChangeType {
	export const Created = 1;
	export const Changed = 2;
	export const Deleted = 3;
}
```

Reply:
Well, spotted.
I've also noticed this while refactoring this (can't decide how to write that code since there is the LSP-side and API-side that should likely use different types...)

Reply:
I saw that you mainly used 4, because it is convenient to use with `|=`.

IMO I think that LSP should follow the LSP spec, thus I would suggest `3`.
Plus, I would try to avoid any confusion latter for DELETE somewhere being 3 and somewhere being 4. 
But you decide. :)

Reply:
I just didn't notice that the kind used when registering the watcher was a bit field while on notifying was just a number.

Reply:
BTW. This is fixed.